lib/sycamore/sycamore/tests/integration/llms/test_openai.py

```diff
@@ -87,12 +99,45 @@ def test_cached_openai_mismatch(tmp_path: Path):
     key = llm._get_cache_key(prompt_kwargs, {})
 
     # store a modified result in the cache (changed prompt_kwargs), ensure there is a cache miss
-    custom_output = {"result": "This is a custom response", "prompt_kwargs": {}, "llm_kwargs": {}}
+    custom_output = {
+        "result": "This is a custom response",
+        "prompt_kwargs": {},
+        "llm_kwargs": {},
+        "model_name": "gpt-3.5-turbo",
+    }
     cache.set(key, custom_output)
 
     assert llm.generate(prompt_kwargs=prompt_kwargs, llm_kwargs={}) != custom_output["result"]
 
 
+def test_cached_openai_different_models(tmp_path: Path):
+    cache = DiskCache(str(tmp_path))
+    llm_GPT_3_5_TURBO = OpenAI(OpenAIModels.GPT_3_5_TURBO, cache=cache)
+    llm_GPT_4O_MINI = OpenAI(OpenAIModels.GPT_4O_MINI, cache=cache)
+
+    prompt_kwargs = {"prompt": "Write a limerick about large language models."}
+    key_GPT_3_5_TURBO = llm_GPT_3_5_TURBO._get_cache_key(prompt_kwargs, {})
+    res_GPT_3_5_TURBO = llm_GPT_3_5_TURBO.generate(prompt_kwargs=prompt_kwargs, llm_kwargs={})
+
+    # assert result is cached
+    assert cache.get(key_GPT_3_5_TURBO).get("result") == res_GPT_3_5_TURBO
+    assert cache.get(key_GPT_3_5_TURBO).get("prompt_kwargs") == prompt_kwargs
+    assert cache.get(key_GPT_3_5_TURBO).get("llm_kwargs") == {}
+    assert cache.get(key_GPT_3_5_TURBO).get("model_name") == "gpt-3.5-turbo"
+
+    key_GPT_4O_MINI = llm_GPT_4O_MINI._get_cache_key(prompt_kwargs, {})
```

Comment:
you can move this (cache population) before the retrievals (checks). I don't think this test is checking if the key is being overwritten by gpt 4?